### DISCUSSION TITLE
Slow chunk generation issues

### DISCUSSION THREAD

**Discussion by nmills3** [itzg/docker-minecraft-server#1412] [General]
### Enhancement Type

Not sure

### Describe the enhancement

I'm running my server on an unraid pc with a 5700g on a 860 evo ssd and with 10gb of ram allocated to it but for some reason i'm not even getting fast enough world generation to be able to fly around in creative. This is with a single player flying around with the server view distance set to 10 chunks. i've also been trying to pre-generate the world but even that runs slowly at around 2/3 chunks per second. Is there anything i can do to improve this performance or is this just how it normally is for everyone?
running spigot 1.18.2 btw

**Comment by null-domain** [comment]
I'm having the same issue on Paper 1.19.2 with 6GB RAM allocated. It worked perfectly on my Windows machine in Docker Desktop (same resource allocation), but when I transferred the files over to my server and ran it there, I started getting incredibly slow chunk generation.

**Reply by itzg** [reply]
@null-domain I'm wondering if the particular filesystem on the newer server is slow for some reason.

**Reply by null-domain** [reply]
Unfortunately I'm still encountering this issue. Brand new server, 6 vCPU cores and 16GB of RAM with a 200 GB NVMe drive. I allocated 12G of mem to the MC server, have enabled autopause and disabled watchdog with a max_tick_time of -1.

**Reply by itzg** [reply]
Be sure to also contact Paper forums/discussion. Containerization has no impact on CPU, memory, disk performance -- containers are just Linux processes.

**Reply by LeonZewe** [reply]
@null-domain I was able to fix this exact issue by reformatting my cache drive (which the server is running on) to xfs instead of btrfs, hope it helps in case you're still having this issue.
@itzg You might wanna consider adding a hint regarding this somewhere in the documentation, I'm thinking theres a good chance other users might face this issue as well.

**Comment by xslugx** [comment]
> @null-domain I was able to fix this exact issue by reformatting my cache drive (which the server is running on) to xfs instead of btrfs, hope it helps in case you're still having this issue. @itzg You might wanna consider adding a hint regarding this somewhere in the documentation, I'm thinking theres a good chance other users might face this issue as well.

Make a backup of the drive, then reformat as xfs, copy backup back to drive?

**Reply by LeonZewe** [reply]
Thats one way to go about it, yup

**Reply by xslugx** [reply]
From what I can see my drive is formatted as fat32 per the unraid instructions. all other drives show xfs formatting

**Reply by xslugx** [reply]
I think I found the issue, I need to manage my cache pool with my SSD's, this is going to be fun

**Comment by geeth24** [comment]
is there a proper fix for this?

**Reply by xslugx** [reply]
I did not have any success even setting a cache pool. It was still super laggy, however I was running ATM 8 mod, which is kinda slow generation anyway

**Reply by geeth24** [reply]
hm, i just put the app on my cache ssd and it got a bit better

**Comment by Venipa** [comment]
i would suggest increasing swap size and set a static value for the docker stack
